### DISCUSSION TITLE
Cannot use the "Poextrator" in Orchard Core 1.4 the generate the PO file for Localization.

### DISCUSSION THREAD

**Discussion by samuel-tw** [OrchardCMS/OrchardCore#12612] [Q&A]
Hi:
I have build a site with Orchard Core 1.4 with localization feature. And according the document,https://docs.orchardcore.net/en/dev/docs/reference/modules/Localize/, I try to use the Poextractor.OrchardCore with no success. Thanks I similar issues on the thread https://github.com/lukaskabrt/PoExtractor/issues/22. It may come from mismatch of the version between poextractor and Orchard Core version. Would you please which version of poextracotor can generate the Orchard Core 1.4 file?
If not, is there any way (mannually?) or guidline to genertate the PO files for Orchard Core 1.4? Please kindly advise.
Here is the error message when I use poextractor to Orchard Core 1.4

` extractpo-oc /Users/samuelliu/Projects/OrchardCore/src/OrchardCore.Themes  /Users/samuelliu/Projects/temp/OrchardCore  --liquid    
Unhandled exception. System.MissingMethodException: Method not found: 'System.Threading.Tasks.Task`1<Fluid.Values.FluidValue> Fluid.Ast.Expression.EvaluateAsync(Fluid.TemplateContext)'.
   at PoExtractor.Liquid.LiquidStringExtractor.TryExtract(LiquidExpressionContext expressionContext, LocalizableStringOccurence& result)
   at PoExtractor.Liquid.ExtractingLiquidWalker.ProcessFilterExpression(FilterExpression filter) in D:\a\1\s\PoExtractor.Liquid\ExtractingLiquidWalker.cs:line 110
   at PoExtractor.Liquid.ExtractingLiquidWalker.Visit(Expression expression) in D:\a\1\s\PoExtractor.Liquid\ExtractingLiquidWalker.cs:line 96
   at PoExtractor.Liquid.ExtractingLiquidWalker.Visit(Statement node)
   at PoExtractor.Liquid.ExtractingLiquidWalker.Visit(Statement node) in D:\a\1\s\PoExtractor.Liquid\ExtractingLiquidWalker.cs:line 69
   at PoExtractor.Liquid.LiquidProjectProcessor.Process(String path, String basePath, LocalizableStringCollection strings) in D:\a\1\s\PoExtractor.Liquid\LiquidProjectProcessor.cs:line 38
   at PoExtractor.OrchardCore.Program.Main(String[] args) in D:\a\1\s\PoExtractor.OrchardCore\Program.cs:line 52
zsh: abort      extractpo-oc /Users/samuelliu/Projects/OrchardCore/src/OrchardCore.Themes  
`

**Comment by hishamco** [comment]
@lukaskabrt are you still maintaing the repo, if yes could you please let us know if this issue related to the tool or not? If Yes we can file an issue in PoExtractor repo

**Comment by samuel-tw** [comment]
Yes, I am still maintain the repo and the issue is considered related to the tool PoExtractor. So I change the Category setting from general to Q&A now. Thanks for your advise.

**Reply by hishamco** [reply]
> Yes, I am still maintain the repo and the issue is considered related to the tool PoExtractor

I was referring to Lukas ;) because he can save my time if it's related to the tool, otherwise we might have a look

**Reply by Skrypt** [reply]
I asked him the same question a while ago on the PoExtractor repository but never had an answer. I asked if there was any progress on migrating the tool to .NET 6.0 and nothing. I suggest we create a fork of the tool and add it in the Orchard Organization before it disapears. @sebastienros

**Reply by hishamco** [reply]
We can moved the repo into Orchard Core organization or I can add it as a part of Orchard Core Contribute then I will be ready to become the maintainer, because I already did some work on the extraction long time ago

**Comment by lukaskabrt** [comment]
Hi, sorry for late answer. I am no longer using the library, so unfortunately, I have very little time to maintain it. If it is OK with you, I would be happy to transfer it under Orchard Organization.

**Reply by hishamco** [reply]
Thanks for your contribution Lukas, let wait for @sebastienros thought either:

- Move it to Orchard Core Organization as a separated repo
- Move it to Orchard Core repo
- Move it to Orchard Core Contrib repo as a community project

I never mind where it goes, but the important thing is someone to maintain it

**Reply by hishamco** [reply]
FYI if this will take time to take a proper decision, I will fix the current issues in my fork because we don't need to block others

**Reply by hishamco** [reply]
We successfuly transfer the repo to OCC as initiative to accelarate the process and fix all `PoExtractor` related issues

**Comment by hishamco** [comment]
@samuel-tw we already use `PoExtractor` for `OC.Translations`, could you please try it on the entire source. I will push a newer version of `PoExtractor` in the upcoming days, I'm trying to fix the current related issues as much as I can
